Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -1201,129 +911,121 @@ private bool MatchesPublishedServerUrl(IPObject source, bool isInExternalSubnet,
-                _logger.LogDebug("{Source}: GetBindInterface: Selected first user defined interface. {Result}", source, result);
-
-                if (isInExternalSubnet)
+                else
                 {
-                    _logger.LogWarning("{Source}: External request received, however, only an internal interface bind found.", source);
+                    // Check to see if any of the internal bind interfaces are in the same subnet as the source.
+                    // If none exists, this will select the first internal interface if there is one.
+                    bindAddress = _interfaces.Where(x => IsInLocalNetwork(x.Address))
+                        .OrderByDescending(x => x.Subnet.Contains(source))
+                        .ThenBy(x => x.Index)
+                        .Select(x => x.Address)
+                        .FirstOrDefault();
+
+                    if (bindAddress != null)
+                    {
+                        result = NetworkExtensions.FormatIpString(bindAddress);
+                        _logger.LogDebug("{Source}: Internal request received, matching internal bind address found: {Result}", source, result);
+                        return true;
+                    }
                 }
-
-                return true;
             }
 
             return false;
         }
 
         /// <summary>
-        /// Attempts to match the source against an external interface.
+        /// Attempts to match the source against external interfaces.
         /// </summary>
         /// <param name="source">IP source address to use.</param>
         /// <param name="result">The result, if a match is found.</param>
         /// <returns><c>true</c> if a match is found, <c>false</c> otherwise.</returns>
-        private bool MatchesExternalInterface(IPObject source, out string result)
+        private bool MatchesExternalInterface(IPAddress source, out string result)
         {
             result = string.Empty;
             // Get the first WAN interface address that isn't a loopback.
-            var extResult = _interfaceAddresses
-                .Exclude(_bindExclusions, false)
-                .Where(p => !IsInLocalNetwork(p))
-                .OrderBy(p => p.Tag)
-                .ToList();
+            var extResult = _interfaces.Where(p => !IsInLocalNetwork(p.Address)).OrderBy(x => x.Index);
 
-            if (extResult.Any())
+            IPAddress? hasResult = null;
+            // Does the request originate in one of the interface subnets?
+            // (For systems with multiple internal network cards, and multiple subnets)
+            foreach (var intf in extResult)
             {
-                // Does the request originate in one of the interface subnets?
-                // (For systems with multiple internal network cards, and multiple subnets)
-                foreach (var intf in extResult)
+                hasResult ??= intf.Address;
```

Comment:
This will only assign to hasResult if hasResult is null. Is that what you want? And if so, why?